extensions/wikia/CommunityPage/scripts/firstEditModal.js

```diff
@@ -0,0 +1,130 @@
+require([
+	'jquery',
+	'wikia.ui.factory',
+	'wikia.mustache',
+	'communitypage.templates.mustache',
+	'wikia.nirvana',
+	'wikia.tracker',
+	'wikia.window',
+	'wikia.cookies',
+], function ($, uiFactory, mustache, templates, nirvana, tracker, window, cookies) {
+	'use strict';
+
+	 var track = tracker.buildTrackingFunction({
+		 action: tracker.ACTIONS.CLICK,
+		 category: 'community-page-first-edit',
+		 trackingMethod: 'analytics'
+	 });
+
+	function getUiModalInstance() {
+		var $deferred = $.Deferred();
+
+		uiFactory.init(['modal']).then(function (uiModal) {
+			$deferred.resolve(uiModal);
+		});
+
+		return $deferred;
+	}
+
+	function getModalContents() {
+		var $deferred = $.Deferred();
+
+		nirvana.sendRequest({
+			controller: 'CommunityPageSpecial',
+			method: 'getFirstTimeEditorModalData',
+			data: {
+				uselang: window.wgUserLanguage
+			},
+			format: 'json',
+			type: 'get',
+		}).then(function (response) {
+			$deferred.resolve(response);
+		});
+
+		return $deferred;
```

Comment:
Likewise, since you're not manipulating the response before resolving, couldn't this just return the `sendRequest` as that itself returns a Promise?